test/minikube/minikube_domain_resync_test.go

```diff
@@ -0,0 +1,193 @@
+		}
+
+		if err != nil {
+			return
+		}
+
+		processes = append(processes, obj)
+	}
+}
+
+func checkArchives(t *testing.T, namespaceName string, adminPod string, numExpected int, numExpectedRemoved int) (archives []archive, removedArchives []archive) {
+	options := k8s.NewKubectlOptions("", "")
+	options.Namespace = namespaceName
+
+	// check archives
+	output, err := k8s.RunKubectlAndGetOutputE(t, options, "exec", adminPod, "--",
+		"nuocmd", "--show-json", "get", "archives", "--db-name", "demo")
+	assert.NilError(t, err, output)
+
+	err, archives = unmarshalArchives(output)
+	assert.NilError(t, err)
+	assert.Equal(t, numExpected, len(archives), output)
+
+	// check removed archives
+	output, err = k8s.RunKubectlAndGetOutputE(t, options, "exec", adminPod, "--",
+		"nuocmd", "--show-json", "get", "archives", "--db-name", "demo", "--removed")
+	assert.NilError(t, err, output)
+
+	err, removedArchives = unmarshalArchives(output)
+	assert.NilError(t, err)
+	assert.Equal(t, numExpectedRemoved, len(removedArchives), output)
+	return
+}
+
+func TestDomainResync(t *testing.T) {
+	testlib.AwaitTillerUp(t)
+	defer testlib.VerifyTeardown(t)
+	defer testlib.Teardown(testlib.TEARDOWN_ADMIN)
+
+	helmChartReleaseName, namespaceName := testlib.StartAdmin(t, &helm.Options{}, 1, "")
+
+	admin0 := fmt.Sprintf("%s-nuodb-cluster0-0", helmChartReleaseName)
+
+	defer testlib.Teardown(testlib.TEARDOWN_DATABASE) // ensure resources allocated in called functions are released when this function exits
+
+	testlib.StartDatabase(t, namespaceName, admin0, &helm.Options{
+		SetValues: map[string]string{
+			"database.sm.resources.requests.cpu":    "0.25",
+			"database.sm.resources.requests.memory": testlib.MINIMAL_VIABLE_ENGINE_MEMORY,
+			"database.te.resources.requests.cpu":    "0.25",
+			"database.te.resources.requests.memory": testlib.MINIMAL_VIABLE_ENGINE_MEMORY,
+		},
+	})
+
+	originalArchiveVolumeClaims := verifyProcessLabels(t, namespaceName, admin0)
+	assert.Equal(t, 1, len(originalArchiveVolumeClaims))
+	originalArchiveId := ""
+	for _, archiveId := range originalArchiveVolumeClaims {
+		originalArchiveId = archiveId
+	}
+	assert.Assert(t, originalArchiveId != "")
+
+	// update replica count
+	options := k8s.NewKubectlOptions("", "")
+	options.Namespace = namespaceName
+	output, err := k8s.RunKubectlAndGetOutputE(t, options, "get", "statefulset", "-o", "jsonpath={.items[*].metadata.name}")
+	assert.NilError(t, err, output)
+
+	statefulSets := strings.Split(output, " ")
+	assert.Equal(t, 3, len(statefulSets), "Expected 3 StatefulSets: Admin, SM, and hotcopy SM")
+
+	// by default the hotcopy SM replica count is 1 and regular SM count is 0
+	// scale regular SM replica count up to 1
+	smStatefulSet := ""
+	for _, statefulSet := range statefulSets {
+		if strings.HasPrefix(statefulSet, "sm-") && !strings.Contains(statefulSet, "hotcopy") {
+			k8s.RunKubectl(t, options, "scale", "statefulset", statefulSet, "--replicas=1")
+			smStatefulSet = statefulSet
+		}
+	}
```

Comment:
you can achieve this by giving Helm the options for `nuodb.database.sm.noHotCopy.Replicas` or similar around lines 126

Reply:
Makes sense. That could cut down test time a bit. But it would make it harder to figure out which PVC corresponds to the scaled down StatefulSet (hotcopy). Currently the test causes PVCs to be created in an orderly fashion so that we know which is which.